Validation/EcalDigis/interface/EcalSelectiveReadoutValidation.h

```diff
@@ -7,47 +7,32 @@
  *
  */
 
-#include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/EDAnalyzer.h"
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+
+#include "DQMServices/Core/interface/MonitorElement.h"
 
-#include "DataFormats/Common/interface/Handle.h"
 #include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
 #include "SimDataFormats/CaloHit/interface/PCaloHit.h"
 
-
-#include "Geometry/CaloTopology/interface/EcalTrigTowerConstituentsMap.h"
-#include "DataFormats/EcalDetId/interface/EcalScDetId.h"
-#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
-
 #include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
-//#include "DataFormats/EcalDetId/interface/EEDetId.h"
-//#include "DataFormats/EcalDetId/interface/EBDetId.h"
 #include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
-#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
 #include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/EcalDetId/interface/EcalScDetId.h"
+#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
 
 #include "Validation/EcalDigis/src/CollHandle.h"
 
 #include <string>
 #include <set>
-#include <utility>
 #include <fstream>
-#include <inttypes.h>
-#include "DQMServices/Core/interface/DQMStore.h"
-#include "DQMServices/Core/interface/MonitorElement.h"
 
-class DQMStore;
-class MonitorElement;
 class EBDetId;
 class EEDetId;
-class EcalTPParameters;
 class EcalElectronicsMapping;
+class EcalTrigTowerConstituentsMap;
 
-class EcalSelectiveReadoutValidation: public edm::EDAnalyzer{
+class EcalSelectiveReadoutValidation: public DQMEDAnalyzer{
```

Comment:
The problem is
http://cmslxr.fnal.gov/lxr/source/Validation/EcalDigis/src/EcalSelectiveReadoutValidation.cc?v=CMSSW_7_5_X_2015-05-11-2300#1637

Which could become thread safe by doing
```
static std::atomic<bool> firstCall {true};
bool expected = true;
if( firstCall.compare_exchange_string(expected, false) ) {
   ...
}
```
This will call the block if and only if this is the thread that changed ```firstCall``` from ```true``` to ```false```.